src/models/krate.rs

```diff
@@ -501,6 +504,11 @@ mod tests {
         assert!(Crate::valid_feature("c++20"));
         assert!(Crate::valid_feature("krate/c++20"));
         assert!(!Crate::valid_feature("c++20/wow"));
+        assert!(Crate::valid_feature("foo?/bar"));
+        assert!(Crate::valid_feature("dep:foo"));
+        assert!(!Crate::valid_feature("dep:foo?/bar"));
```

Comment:
Why is this not valid? `dep:foo/bar` should also be valid IMO

Reply:
We decided not to allow combining the the two syntaxes to simplify it.  There is more information at https://github.com/rust-lang/cargo/pull/9574 about that change.